### PR TITLE
Merge NSSL branch v8merge into the MPAS GSL-develop

### PR DESCRIPTION
**Pull request by tanyasmirnova**
The title above should be a 1 line short summary of the pull request (i.e. what the project the PR represents is intended to do).

Merged NSSL code to the GSL MPAS branch.

Enter a description of this PR. This should include why this PR was created, and what it does.
This merge adds RUC LSM and NSSL microphysics, initial/BC conditions for all hydrometeors, and more diagnostics.

Testing and relations to other Pull Requests should be added as subsequent comments.

See the below examples for more information.
https://github.com/MPAS-Dev/MPAS/pull/930
https://github.com/MPAS-Dev/MPAS/pull/931

### CODE REVIEW

File: src/core_atmosphere/dynamics/mpas_atm_time_integration.F
Comment (joeolson42) [inline src/core_atmosphere/dynamics/mpas_atm_time_integration.F:1334]: Note to Anders: i guess ifa & wfa need to be added here too

File: src/core_atmosphere/physics/mpas_atmphys_control.F
Comment (joeolson42) [inline src/core_atmosphere/physics/mpas_atmphys_control.F:387]: @tanyasmirnova : is this really correct? the 9th layer is thinner than the 8th?

File: src/core_atmosphere/physics/mpas_atmphys_manager.F
Comment (joeolson42) [inline src/core_atmosphere/physics/mpas_atmphys_manager.F:699]: @AndersJensen-NOAA : do we need anything like this for TEMPO?

File: src/core_atmosphere/physics/mpas_atmphys_control.F
Comment (tanyasmirnova) [reply src/core_atmosphere/physics/mpas_atmphys_control.F:387]: @joeolson42 Joe, these are the thicknesses between the middles of the layers. The first and last values are actually 0.5 of the thickness.

File: src/core_atmosphere/Registry.xml
Comment (AndersJensen-NOAA) [inline src/core_atmosphere/Registry.xml:1552]: I added this in my PR.
Setting `packages="mp_thompson_in;mp_nssl2m_in"` will solve the merge conflict for this.

**Review by joeolson42 — APPROVED** [review]
I left a few comments. I don't know much of anything about the majority of the modules changed, so my review isn't worth much. Under these time constraints, we have no choice but to accept the changes.

File: src/core_atmosphere/physics/mpas_atmphys_control.F
Comment (joeolson42) [reply src/core_atmosphere/physics/mpas_atmphys_control.F:387]: OK. Thanks.

File: src/core_atmosphere/dynamics/mpas_atm_time_integration.F
Comment (joeolson42) [reply src/core_atmosphere/dynamics/mpas_atm_time_integration.F:1334]: I meant to tag @AndersJensen-NOAA

**Review by mdtoyNOAA — APPROVED** [review]

File: src/core_atmosphere/diagnostics/Registry_convective.xml
Comment (AndersJensen-NOAA) [inline src/core_atmosphere/diagnostics/Registry_convective.xml:1]: Can we ignore this registry for convective diagnostics?

File: src/core_atmosphere/diagnostics/mpas_atm_diagnostics_manager.F
Comment (AndersJensen-NOAA) [inline src/core_atmosphere/diagnostics/mpas_atm_diagnostics_manager.F:1]: These changes could be ignored.

File: src/core_atmosphere/diagnostics/mpas_convective_diagnostics.F
Comment (AndersJensen-NOAA) [inline src/core_atmosphere/diagnostics/mpas_convective_diagnostics.F:1]: These changes could be ignored.

File: src/core_atmosphere/dynamics/mpas_atm_time_integration.F
Comment (AndersJensen-NOAA) [reply src/core_atmosphere/dynamics/mpas_atm_time_integration.F:1334]: Yes, @joeolson42, I have ifa and nwfa and nc added in my PR.

File: src/core_atmosphere/dynamics/mpas_atm_time_integration.F
Comment (AndersJensen-NOAA) [inline src/core_atmosphere/dynamics/mpas_atm_time_integration.F:6426]: The changes above Line 6426 are formatting changes and adding hydrometeors to the boundary conditions, so we can keep those changes. Changes below line 6426 should be understood before merging.

File: src/core_atmosphere/physics/Makefile
Comment (AndersJensen-NOAA) [inline src/core_atmosphere/physics/Makefile:1]: These look good to keep.

File: src/core_atmosphere/mpas_atm_core.F
Comment (AndersJensen-NOAA) [inline src/core_atmosphere/mpas_atm_core.F:1]: These changes could be ignored.

File: src/core_atmosphere/physics/mpas_atmphys_driver_cloudiness.F
Comment (AndersJensen-NOAA) [inline src/core_atmosphere/physics/mpas_atmphys_driver_cloudiness.F:1]: Ignore this change

File: src/core_atmosphere/physics/mpas_atmphys_driver.F
Comment (AndersJensen-NOAA) [inline src/core_atmosphere/physics/mpas_atmphys_driver.F:273]: We should only add changes specific to the LSM from this file.

File: src/core_atmosphere/physics/mpas_atmphys_driver.F
Comment (AndersJensen-NOAA) [inline src/core_atmosphere/physics/mpas_atmphys_driver.F:280]: We should only add changes specific to the LSM from this file.

File: src/core_atmosphere/physics/mpas_atmphys_driver_microphysics.F
Comment (AndersJensen-NOAA) [inline src/core_atmosphere/physics/mpas_atmphys_driver_microphysics.F:1]: Do not merge changes from this file (the microphysics driver).

File: src/core_atmosphere/physics/mpas_atmphys_manager.F
Comment (AndersJensen-NOAA) [reply src/core_atmosphere/physics/mpas_atmphys_manager.F:699]: @joeolson42, likely. Though we'll have to look at timestep sensitivity in MPAS first. There already is something defined in the lines above for Thompson, a limit of 90 s.

File: src/core_atmosphere/physics/mpas_atmphys_vars.F
Comment (AndersJensen-NOAA) [inline src/core_atmosphere/physics/mpas_atmphys_vars.F:252]: Set to false before merge `f_qh = .false.`

File: src/core_atmosphere/physics/mpas_atmphys_constants.F
Comment (AndersJensen-NOAA) [inline src/core_atmosphere/physics/mpas_atmphys_constants.F:99]: a2, a3, and a4 replace a2s, a3s, and a4s defined as parameters in the scheme itself.

File: src/core_atmosphere/physics/physics_wrf/module_mp_thompson.F
Comment (AndersJensen-NOAA) [inline src/core_atmosphere/physics/physics_wrf/module_mp_thompson.F:1]: Do not merge changes to this file: `module_mp_thompson.F`

File: src/core_init_atmosphere/Makefile
Comment (AndersJensen-NOAA) [inline src/core_init_atmosphere/Makefile:1]: Since some of these commit are old, I'd ignore changes to the Makefile.

File: src/driver/Makefile
Comment (AndersJensen-NOAA) [inline src/driver/Makefile:1]: Ignore this.

File: src/framework/Makefile
Comment (AndersJensen-NOAA) [inline src/framework/Makefile:1]: Ignore

**Review by AndersJensen-NOAA — CHANGES_REQUESTED** [review]
The commits from Michael Duda before 2021 are related to the use of SMIOL, which is a built-in I/O layer. If we don't intent on using that, then we can ignore those commits. There are also a lot of commits related to the addition of severe convection diagnostics, that we could ignore for now as well. I added suggestions for files to ignore; here's the caveat: I don't what will or won't break if we ignore files that I suggest we ignore.

**Comment by tanyasmirnova** [comment]
> I left a few comments. I don't know much of anything about the majority of the modules changed, so my review isn't worth much. Under these time constraints, we have no choice but to accept the changes.

@joeolson42 Joe, I think we could make a separate branch: NCAR_NSSLv8_merge. And another option isto add our physics to NCAR MPAS one at a time as you suggested. I am working now on separating RUC LSM changes from the rest of NSSL changes. Hopefully it will not take too long, maybe by next week it will be ready. The NCAR_NSSLv8_merge branch could be used untill we are ready with the NCAR_GSL/develop.

**Comment by jderrico-noaa** [comment]
> @joeolson42 Joe, I think we could make a separate branch: NCAR_NSSLv8_merge. And another option isto add our physics to NCAR MPAS one at a time as you suggested. I am working now on separating RUC LSM changes from the rest of NSSL changes. Hopefully it will not take too long, maybe by next week it will be ready. The NCAR_NSSLv8_merge branch could be used until we are ready with the NCAR_GSL/develop.

I think this makes sense, I can add @tanyasmirnova 's branch from this PR as a separate branch to our repo for now and then work on pulling out what @AndersJensen-NOAA flagged and we can see how (if) it compiles and runs.  In the mean time we can start with Anders's PR #2, I need one more approval and then I'll merge it

**Comment by tanyasmirnova** [comment]
@jderrico-noaa @joeolson42 @AndersJensen-NOAA This merged with NSSL version of MPAS code has been tested on the 13-km CONUS test case from Anders Jensen. Although the versions of MYNN, Thompson and RUC LSM are old, the results appear to be reasonable. Below is the 2-m T verification produced by @AndersJensen-NOAA:
image (1)
This code could be used in HFIP as a backup until the gsl/develp is ready.

**Pull request closed without merging**
